magma/lib/magma/model.rb

```diff
@@ -26,6 +26,9 @@ def load_attributes(attributes = [])
             alias_method "#{attribute.attribute_name}=", "#{attribute.column_name}="
           end
         end
+
+        # Table models should always have an identifier attribute
+        self.attributes[self.identity.name] = self.identity unless attributes.include?(self.identity)
```

Comment:
This is required because for some reason, table models weren't getting the `id` loaded initially, so first queries always returned no `id` field. Subsequent queries worked, so that was getting added in somewhere in the loop. We should just enforce that table models always load with an identifier attribute.

Reply:
Hmm, the logic around this bit of weirdness is very warped, tread carefully here.